### ISSUE TITLE
no-unused-expressions false-positive with ternary operator

### ISSUE THREAD

**Issue by vlmonk** [eslint/eslint#2181]
Hi. I guess `no-unused-expressions` rule is false-positive in this test-case

```js
"use strict";
var fs = require("fs");

function test () {
  return new Promise(function (resolve, reject) {
    fs.readFile("file.txt", function(err, data) {
      err ? reject(err) : resolve(data);
    });
  });
}

test();
```

I got error with default config: 

```
eslint --env node test.js

test.js
  7:6  error  Expected an assignment or function call and instead saw an expression  no-unused-expressions
```

**Comment by nzakas** [comment]
The rule is correct. The ternary operator is used to return a value based on a condition. In your example, you are throwing away the return value, as if doing `5+5;`, so it is being flagged. You can either change your code to use a regular `if` statement or disable the rule around this exception using `//ESLint - disable - line. See http://eslint.org/docs/user-guide/configuring
